### DISCUSSION TITLE
renovate config validator steps to run

### DISCUSSION THREAD

**Discussion by shaliniC1706** [renovatebot/renovate#8113] [Request Help]
I have updated the renovate.json via raising a PR on git hub now how to run renovate config validator to confirm before merging?

**Comment by viceice** [comment]
check this: https://github.com/renovatebot/docker-renovate#config-validator

**Reply by shaliniC1706** [reply]
Thanks for the quick reply. I have changed the base branch in onboarding PR but it is not raising any PR for the the new base branch. Any Idea why?

**Reply by viceice** [reply]
check the logs?

**Reply by shaliniC1706** [reply]
In the logs it is still trying to default branch master

**Reply by viceice** [reply]
Please create a minimal public reproduction repo on github, so we can check.

Please post the logs here after run.

**Reply by shaliniC1706** [reply]
Is this necessary to merge the onboarding PR in to master only?

**Reply by viceice** [reply]
sure, renovate will only use the config from `default` branch

**Reply by viceice** [reply]
you can use the config validator in a github action running on you pr, so it will validate your new config

**Reply by shaliniC1706** [reply]
Is this a part of config?

**Comment by rarkins** [comment]
Renovate config must be in the default branch (e.g. `master` or `main`). Although you can define multiple or alternative `baseBranches` in that config file, any config file which you have in those base branches themselves will be ignored.

**Reply by shaliniC1706** [reply]
Ok, so I can only change the baseBranches in config of default branch. That implies first time onboard required merging in master branch right? I was trying to test it on test branch first.

**Reply by rarkins** [reply]
I *think* that if you add a `baseBranches` setting to the `renovate.json` in your onboarding PR that it should update to reflect those base branches, but I'm not 100% certain of it. i.e. you can still get a preview

**Reply by shaliniC1706** [reply]
OK thanks for the help.

**Reply by shaliniC1706** [reply]
{
  "extends": [
    "config:base",
    ":labels(depedendencies)"
  ],
  "branchPrefix":"renovate_",
  "baseBranches":["US15314-Update_deps"]
}

will this configuration update the branch prefix as "renovate_" instead of renovate/ ? also please confirm if the base branch config is right or not?

**Reply by rarkins** [reply]
Yes and Yes

**Reply by shaliniC1706** [reply]
ok thanks

**Reply by shaliniC1706** [reply]
also I need to update the renovate config to only allow bazel manager. Is it possible?

**Reply by rarkins** [reply]
https://docs.renovatebot.com/configuration-options/#enabledmanagers

Also note that the base branch must already exist otherwise it's skipped with a warning

**Reply by shaliniC1706** [reply]
Ok Thanks. If I change something in my base branch renovate.json, will this change reflects on my future PR?

**Reply by rarkins** [reply]
Yes

**Reply by shaliniC1706** [reply]
I think it's ignoring the base branch renovate.json and only took default branch's renovate.json. Also the config validation triggers only when there is a change in default branch's renovate.json. Is it expected?

**Reply by viceice** [reply]
Yes, renovate always only reads config from default branch
